iidm/iidm-api/src/main/java/com/powsybl/iidm/network/ValidationUtil.java

```diff
@@ -75,23 +186,13 @@ public static void checkTargetDeadband(Validable validable, String validableType
         }
     }
 
-    public static boolean checkVoltageControl(Validable validable, Boolean voltageRegulatorOn, double voltageSetpoint) {
-        if (voltageRegulatorOn == null) {
-            throw new ValidationException(validable, "voltage regulator status is not set");
-        }
-        boolean bVoltageRegulatorOn = voltageRegulatorOn; // make sonar happy java:S5411 Boxed "Boolean" should be avoided in boolean expressions
-        if (bVoltageRegulatorOn) {
-            if (Double.isNaN(voltageSetpoint) || voltageSetpoint <= 0) {
-                throw createInvalidValueException(validable, voltageSetpoint, "voltage setpoint", "voltage regulator is on");
-            }
-            return false;
-        }
-        return true;
-    }
-
-    public static void checkVoltageControl(Validable validable, Boolean voltageRegulatorOn, double voltageSetpoint, double reactivePowerSetpoint) {
-        if (checkVoltageControl(validable, voltageRegulatorOn, voltageSetpoint) && Double.isNaN(reactivePowerSetpoint)) {
-            throw createInvalidValueException(validable, reactivePowerSetpoint, "reactive power setpoint", "voltage regulator is off");
+    /**
+     * For validating target deadband.
+     * If control is enabled target deadband must be defined and valid (>= 0.0).
+     */
+    public static void checkTargetDeadband(Validable validable, String validableType, double targetDeadband, boolean voltageRegulatorOn) {
+        if (voltageRegulatorOn && !validDeadband(targetDeadband)) {
```

Comment:
Not sure if it is worth changing. Even when it is not regulating, having targetDeadband < 0 does not make sense... Before the check was: if regulating, targetDeadband is not NaN (no check if NaN if not regulating) but in every case we must have targetDeadband >=0 if not NaN

Reply:
Changed. When the control is disabled only valid values and NaN (initial value, targetDeadband is not set) are allowed.